tests/test_utils.py

```diff
@@ -134,6 +136,26 @@ def test_patch_environment_key_exists(self):
         assert "BB" not in os.environ
         assert "CC" not in os.environ
 
+    def test_patch_environment_restores_on_error(self):
+        # we need to find an upper-case envvar
+        # because `patch_environment upper-cases all keys...
+        key, orig_value = next(kv for kv in os.environ.items() if kv[0].isupper())
```

Comment:
If there is no upper case env var, this will raise an error. Very unlikely, just mentioning it.